src/app/pages/blog/blog.js

```diff
@@ -3,61 +3,161 @@ import settings from 'settings';
-        this.handlePathChange = () => {
-            if (history && history.state && history.state.model) {
-                Object.assign(this.model, history.state.model);
+    moreStoriesOptions(test) {
+        if (!this.pageData) {
+            return {};
+        }
+        const articles = this.pageData.articles;
+        const slugs = Reflect.ownKeys(articles)
+            .filter((k) => test(k, articles[k]))
+            .sort((a, b) => articles[a].date < articles[b].date ? 1 : -1);
+
+        return slugs.map((s) => blurbModel(s, this.pageData.articles[s]));
+    }
+
+    buildDefaultPage() {
+        const fa = new FeaturedArticle(this.featuredArticleOptions);
+        // const sb = new SearchBar({
+        //     model: {
+        //         title: 'Read more great stories'
+        //     }
+        // });
+        const ub = new UpdateBox({
+            model: {
+                rssUrl: `${settings.apiOrigin}/blog-feed/rss/`
             }
-            const slugMatch = window.location.pathname.match(/\/blog\/(.+)/);
-
-            if (slugMatch) {
-                this.model.articleSlug = slugMatch[1];
-                if (!this.articles[this.model.articleSlug]) {
-                    routerBus.emit('navigate', '/404', {path: '/blog'}, true);
-                    return;
-                }
-                if (this.previousSlug !== slugMatch[1]) {
-                    this.previousSlug = slugMatch[1];
-                    this.setTimers();
-                    this.setCanonicalLink(slugMatch[0], this.canonicalLink);
-                }
+        });
+        const ms = new MoreStories({
+            articles: this.moreStoriesOptions((_, articleEntry) => !articleEntry.pin_to_top)
+        });
+
+        // sb.on('value', (searchParam) => {
+        //     history.pushState({}, '', `${path}/?${searchParam}`);
+        //     this.handlePathChange();
+        // });
+        this.regions.self.attach(fa);
+        // this.regions.self.append(sb);
+        this.regions.self.append(ub);
+        this.regions.self.append(ms);
+    }
+
+    buildSearchResultsPage() {
+        const sb = new SearchBar({
+            model: {
+                title: 'Read more great stories'
+            }
+        });
+        const sr = new SearchResults();
+
+        sb.on('value', (searchParam) => {
+            history.pushState({}, '', `${path}/?${searchParam}`);
+            this.handlePathChange();
+        });
+        this.regions.self.attach(sb);
+        this.regions.self.append(sr);
+    }
+
+    buildArticlePage() {
+        const a = new Article({
+            slug: `news/${this.articleSlug}`
+        });
+        const ub = new UpdateBox({
+            model: {
+                rssUrl: `${settings.apiOrigin}/blog-feed/rss/`
+            }
+        });
+        const ms = new MoreStories({
+            articles: this.moreStoriesOptions((slug, _) => slug !== this.articleSlug)
+        });
+
+        this.regions.self.attach(a);
```

Comment:
Can we just do
```
    this.regions.self.attach(new Article({
            slug: `news/${this.articleSlug}`
        }));
```

Reply:
I'm a little undecided about readability of inlining the component creation, but it's not really bad. Reworked.

Reply:
I just assumed that if we don't  need to access to a in the future, we wouldn't need to declare a new variable for that.